### PR TITLE
added regridding functionality

### PR DESCRIPTION
**Pull request by matt-long**
This includes a preliminary capability to perform regridding based on "named grids". This is only available in Python 3: the regridding module is not loaded when `esmlab` is used within Python 2. 

Future work will be required to enable grid-file generation and post-regridding filling capabilities. 

Additionally, we should consider removing dependencies on xESMF...or pushing functionality into that package.

### CODE REVIEW

File: esmlab/regrid.py [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+    ):
+        """
+        Parameters
+        ----------
+        name_grid_src : string
+                      Name of source grid.
+        name_grid_dst : string
+                      Name of destination grid.
+        method : string, optional
+                Regridding method. Options are:
```
Comment (andersy005) [inline esmlab/regrid.py]: Add an empty line here for bullet points to be properly rendered

Comment (andersy005) [reply esmlab/regrid.py]: This is how they are being rendered: https://339-156243509-gh.circle-artifacts.com/0/html/api.html#esmlab.regrid.regridder

File: esmlab/__init__.py [hunk 2]
```diff
@@ -1,11 +1,18 @@
 #!/usr/bin/env python
+import sys
+from .config import set_options, get_options
```
Comment (andersy005) [inline esmlab/__init__.py]: Can you move these two lines underneath the module doc-string line?

Comment (matt-long) [reply esmlab/__init__.py]: Done.

File: esmlab/regrid.py (hunk 1 above)
Comment (matt-long) [reply esmlab/regrid.py]: done.

Comment (andersy005) [reply esmlab/regrid.py]: Thank you!

**Comment by matt-long** [comment]
I'll work on a test.

File: esmlab/regrid.py [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+
+                - 'bilinear'
+                - 'conservative'
+                - 'patch'
+                - 'nearest_s2d'
+                - 'nearest_d2s'
+        overwrite_existing : logical, optional [Default=False]
```
Comment (andersy005) [inline esmlab/regrid.py]: In the doc-string,  can you change `overwriting_existing` type from `logical` to `bool`

File: esmlab/regrid.py [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+        post_method=None,
+        post_method_kwargs=None,
+    ):
+        """Perform regridding on an `xarray.DataArray`.
+
+        Parameters
+        ----------
+        da_in : xr.DataArray
+              The data to regrid
+        renormalize : logical, optional [default=True]
```
Comment (andersy005) [inline esmlab/regrid.py]: Same thing here.. `logical` -> `bool`

File: esmlab/regrid.py [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+                   to use a mapping file computed at the surface at each
+                   successive depth level: setting `renormalize=True` will
+                   ensure correct handling of missing values.
+        interp_coord : dict, optional
+                   Dictionary specifying dimension names an new coordinates;
+                   passed to `xarray.DataArray.interp`.
+                   New coordinate can be a scalar, array-like or DataArray.
+                   If DataArrays are passed as new coordates, their dimensions
+                   are used for the broadcasting.
+        apply_mask : logical, optional [default=False]
```
Comment (andersy005) [inline esmlab/regrid.py]: Same thing here.. `logical` -> `bool`

**Comment by andersy005** [comment]
I am going to be out for the next 2 hours. If this PR is something you need to use ASAP, feel free to merge it. Otherwise, I will take another look when I am back home.

**Pull request merged**